com.microsoft.mrtk.spatialmanipulation/PlacementHub.cs

```diff
@@ -0,0 +1,407 @@
+            var maxAcceleration = Mathf.Max(springForceLimit, 10 * velocity.magnitude / Time.fixedDeltaTime);
+
+            if (accelerationMagnitude > maxAcceleration)
+            {
+                acceleration *= maxAcceleration / accelerationMagnitude;
+            }
+
+            // Apply damping - mathematically, we need e^(-2 * omega * dt)
+            // To compensate for the finite time step, this is split in two equal factors,
+            // one applied before, the other after the spring force
+            // equivalent with applying damping as well as spring force continuously
+            float halfDampingFactor = Mathf.Exp(-springDamping * omega * Time.fixedDeltaTime);
+
+            velocity -= referenceFrameVelocity;  // change to the player's frame of reference before damping
+
+            velocity *= halfDampingFactor;  // 1/2 damping
+            velocity += acceleration * Time.fixedDeltaTime; // integration step of spring force
+            velocity *= halfDampingFactor;  // 1/2 damping
+
+            velocity += referenceFrameVelocity;  // change back to global frame of reference
+
+            rigidBody.velocity = velocity;
+
+            if (applyTorque)
+            {
+                // Torque calculations: same calculation & parameters as for linear velocity
+                // skipping referenceFrameVelocity and springForceLimit which do not exactly apply here
+
+                // implement critically dampened spring force, scaled to mass-independent frequency
+                float angularOmega = Mathf.PI / springTorqueSoftness;  // angular frequency, sqrt(k/m)
+
+                var angularDistance = transform.rotation * Quaternion.Inverse(targetTransform.Rotation);
+                angularDistance.ToAngleAxis(out float angle, out Vector3 axis);
+
+                if (!axis.IsValidVector())
+                {
+                    // ToAngleAxis is numerically unstable, returning NaN axis for near-zero angles
+                    angle = 0;
+                    axis = Vector3.up;
+                }
+
+                if (angle > 180f)
+                {
+                    angle -= 360f;
+                }
+
+                var angularVelocity = rigidBody.angularVelocity;
+
+                var angularAcceleration = -angle * angularOmega * angularOmega;  // acceleration caused by spring force
+
+                angularVelocity *= halfDampingFactor;  // 1/2 damping
+                angularVelocity += angularAcceleration * Time.fixedDeltaTime * Mathf.Deg2Rad * axis.normalized; // integration step of spring force
+                angularVelocity *= halfDampingFactor;  // 1/2 damping
+
+                rigidBody.angularVelocity = angularVelocity;
+            }
+        }
+        #endregion
+    }
+}
+
+/// <summary>
+/// Interface which describes a transformation that is applied to MixedRealityTransform
+/// </summary>
+public interface ITransformation
```

Comment:
Are these intentionally outside the namespace (or is it just a whitespace rendering issue on GH)? Also, could they be inside their own files?

Reply:
I do want to move it into it's own file in the final version. I wanted to vet the core idea of the architecture first though

Reply:
It's been done